tests/units/classes/test.php

```diff
@@ -1245,5 +1250,51 @@ public function testStaticMethod()
 						->isEqualTo(array($return1, $return2, $return3))
 			;
 		}
+
+		public function testNewMockInstance()
+		{
+			$this
+				->if($test = new emptyTest())
+				->then
+					->object($mock = $test->newMockInstance('stdClass'))
```

Comment:
Here we test all the possible call to `newMockInstance` and check that the mock is correct